src/main/scala/com/yotpo/metorikku/utils/FileUtils.scala

```diff
@@ -58,17 +59,28 @@ object FileUtils {
   }
 
   def readConfigurationFile(path: String): String = {
-    val fileContents = Source.fromFile(path).getLines.mkString("\n")
-    val interpolationMap = System.getProperties().asScala ++= System.getenv().asScala
-    StringSubstitutor.replace(fileContents, interpolationMap.asJava)
+    val envAndSystemProperties = System.getProperties().asScala ++= System.getenv().asScala
+    val prefix = envAndSystemProperties.get("CONFIG_FILES_PATH_PREFIX") match {
+      case Some(prefix) => prefix
+      case _ => ""
+    }
+
+    val fileContents = readFileWithHadoop(prefix + path)
+    StringSubstitutor.replace(fileContents, envAndSystemProperties.asJava)
   }
 
-  def readFileWithHadoop(path: String, sparkSession: SparkSession): String = {
+  def readFileWithHadoop(path: String, ss: Option[SparkSession] = None): String = {
+    val sparkSession = ss match {
+      case Some(session) => session
```

Comment:
why passing the session? why not always do getOrCreate?

Reply:
Is that a good practice?